### PR TITLE
Fix #25761: Site map doesn't include home page

### LINKED ISSUE #25761: Site map doesn't include home page
### Preconditions (*)
1. 2.3.3 or 2.3-develop

### Steps to reproduce (*)

1. Go to Stores > Settings > Configuration > Catalog > XML Sitemap > Generation Settings section
2. Change option Enabled to Yes
3. Go to `Marketing > Seo & Search > Site Map`
4. Create a new site map and generate
5. Open new site map in browser

### Expected result (*)
1. Sitemap should include the home page 
Sitemap

### Actual result (*)
1. Sitemap does note include the home page
Sitemap-1

### PR DESCRIPTION
**Pull request by deepaksnair**
### Description (*)

Site map doesn't include home page url issue fixed
### Fixed Issues (if relevant)

1. magento/magento2#25761: Site map doesn't include home page

### Manual testing scenarios (*)

1. Go to Stores > Settings > Configuration > Catalog > XML Sitemap > Generation Settings section
2. Change option Enabled to Yes
3. Go to Marketing > Seo & Search > Site Map
4. Create a new site map and generate
5. Open new site map in browser

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by deepaksnair** [comment]
@magento give me test instance

**Comment by deepaksnair** [comment]
@magento give me 2.4-develop instance

File: app/code/Magento/Sitemap/etc/adminhtml/system.xml [hunk 1]
```diff
@@ -51,7 +51,19 @@
                     <comment>Valid values range from 0.0 to 1.0.</comment>
                 </field>
             </group>
-            <group id="generate" translate="label" type="text" sortOrder="4" showInDefault="1">
+            <group id="store" translate="label" type="text" sortOrder="4" showInDefault="1" showInWebsite="1" showInStore="1">
+                <label>Store Url Options</label>
+                <field id="changefreq" translate="label" type="select" sortOrder="1" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
```
Comment (lbajsarowicz) [inline app/code/Magento/Sitemap/etc/adminhtml/system.xml]: Good practice is to increase `sortOrder` numbers by 10. This way extension developers are still able to extend the grid with their elements.

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Could you provide automated tests to the PR?

File: app/code/Magento/Sitemap/etc/adminhtml/system.xml (hunk 1 above)
Comment (deepaksnair) [reply app/code/Magento/Sitemap/etc/adminhtml/system.xml]: @lbajsarowicz , thanks for reveiwing PR, I have updated the sort orders. Please help to verify.
Also can you please suggest how can I provide automated tests to PR.

**Comment by magento-engcom-team** [comment]
Hi @deepaksnair. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @deepaksnair. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @deepaksnair, here is your [Magento instance](https://i-26445-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26445-2-4-develop.instances.magento-community.engineering/admin_b6fe](https://i-26445-2-4-develop.instances.magento-community.engineering/admin_b6fe)
Login: `2d937ef6` Password: `e0524ea52c83`
Instance will be terminated in up to 3 hours.

**Comment by deepaksnair** [comment]
Hello @lbajsarowicz , I have added automated test cases in last commit, please help to verify PR.
Thanks

File: app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        
+        $this->assertTrue(count($items) == 1);
+        foreach ($items as $item) {
+            $this->assertSame('daily', $item->getChangeFrequency());
+            $this->assertSame('1.0', $item->getPriority());
+        }
+    }
+    
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
```
Comment (lbajsarowicz) [inline app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php]: Use `\PHPUnit\Framework\MockObject\MockObject`

Comment (lbajsarowicz) [reply app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php]: And return type is actually:
`SitemapItemInterfaceFactory|MockObject`

File: app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                $helper = new ObjectManager($this);
+
+                return $helper->getObject(SitemapItem::class, $data);
+            });
+
+        return $itemFactoryMock;
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
```
Comment (lbajsarowicz) [inline app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php]: Use `\PHPUnit\Framework\MockObject\MockObject`
And return type is `ConfigReaderInterface|MockObject`

File: app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace Magento\Sitemap\Test\Unit\Model\ItemProvider;
+
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Sitemap\Model\ItemProvider\ConfigReaderInterface;
+use Magento\Sitemap\Model\ItemProvider\StoreUrl as StoreUrlItemResolver;
+use Magento\Sitemap\Model\SitemapItem;
+use Magento\Sitemap\Model\SitemapItemInterfaceFactory;
+
+class StoreUrlTest extends \PHPUnit\Framework\TestCase
```
Comment (lbajsarowicz) [inline app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php]: Import class.

File: app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+     * test for getItems method
+     */
+    public function testGetItems()
+    {
+        $configReaderMock = $this->getConfigReaderMock();
+        $itemFactoryMock = $this->getItemFactoryMock();
+        $resolver = new StoreUrlItemResolver($configReaderMock, $itemFactoryMock);
+        $items = $resolver->getItems(1);
+        
+        $this->assertTrue(count($items) == 1);
```
Comment (lbajsarowicz) [inline app/code/Magento/Sitemap/Test/Unit/Model/ItemProvider/StoreUrlTest.php]: There is `->assertCount(1, $items)`

File: app/code/Magento/Sitemap/Model/ItemProvider/StoreUrlConfigReader.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * @param ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(ScopeConfigInterface $scopeConfig)
+    {
+        $this->scopeConfig = $scopeConfig;
+    }
+
+    /**
+     * {@inheritdoc}
```
Comment (lbajsarowicz) [inline app/code/Magento/Sitemap/Model/ItemProvider/StoreUrlConfigReader.php]: `@inheritdoc` should not be in braces

File: app/code/Magento/Sitemap/Model/ItemProvider/StoreUrlConfigReader.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        return (string)$this->scopeConfig->getValue(
+            self::XML_PATH_PRIORITY,
+            ScopeInterface::SCOPE_STORE,
+            $storeId
+        );
+    }
+
+    /**
+     * {@inheritdoc}
```
Comment (lbajsarowicz) [inline app/code/Magento/Sitemap/Model/ItemProvider/StoreUrlConfigReader.php]: `@inheritdoc` should not be in braces

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Great job! Only a few things to be changed.

**Comment by deepaksnair** [comment]
Hello @lbajsarowicz , I have resolved all the feedback given by you, please help to review it again.
Thanks.

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6670](https://jira.corp.magento.com/browse/ENGCOM-6670) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @deepaksnair, thank you for your contribution.

Could you fix failing static tests and add this phrase `Store Url Options` into the translation file?

Thanks!

**Comment by lbajsarowicz** [comment]
@deepaksnair  Please fix Static Tests.
I'd love to help you with Automated tests - call me on Community Engineering Slack, please.

**Comment by deepaksnair** [comment]
Hello @lbajsarowicz , @engcom-Alfa , I have resolved the issues in Static Tests also added translations, please help to verify.
Thanks

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6670](https://jira.corp.magento.com/browse/ENGCOM-6670) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**